common/test/test_sshtools.py

```diff
@@ -396,3 +398,62 @@ def test_startSshAgentMissing(self, mockWhich):
         mockWhich.return_value = ''
         with self.assertRaises(MountException):
             self.ssh.startSshAgent()
+
+
+class SSHCopyID(unittest.TestCase):
+    def test_complete_command(self):
+        """Complete generated command"""
+        command = sshtools.sshCopyIdCommand(
+            generic.PRIV_KEY_FILE,
+            'user',
+            'non_existing_host',
+        )
+        self.assertEqual(
+            command,
+            [
+                'ssh-copy-id',
+                '-i',
+                generic.PRIV_KEY_FILE,
+                '-p',
+                '22',
+                'user@non_existing_host'
+            ]
+        )
+
+    def test_default_port(self):
+        """Default port used"""
+        command = sshtools.sshCopyIdCommand(
+            generic.PRIV_KEY_FILE,
+            'user',
+            'non_existing_host',
+        )
+        self.assertEqual(command[4], '22')
+
+    def test_custom_port(self):
+        """Custom (random) port"""
+        custom_port = str(random.randint(23, 128))
+
+        sut = sshtools.sshCopyIdCommand(
+            generic.PRIV_KEY_FILE,
+            'user',
+            'non_existing_host',
+            port=custom_port
+        )
+        self.assertEqual(sut[4], custom_port)
+
+    def test_proxy(self):
+        """Used proxy"""
+        proxy_user = 'non_existing_proxy_user'
+        proxy_host = 'non_existing_proxy_host'
+
+        sut = sshtools.sshCopyIdCommand(
+            generic.PRIV_KEY_FILE,
+            'user',
+            'non_existing_host',
+            proxy_user=proxy_user,
+            proxy_host=proxy_host
+        )
+
+        self.assertTrue(sut[6].startswith(
```

Comment:
@buhtz your update to this test means that we are no longer testing the inclusion of the port for the proxy

Reply:
@buhtz this is different - the proxy command also has its own port value that would be useful to test

Reply:
Yes, I was definitive to radical here. 😈

Reply:
This test brings me to a question: sshCopyIdCommand() do not use a default port value. What should be the expected behavior?

```
def sshCopyIdCommand(
    pubkey,
    user,
    host,
    port='22',
    proxy_user=None,
    proxy_host=None,
    proxy_port=None,
    cipher=None
):
```

I would argument to give `proxy_port` the default value `'22'` because it is the same for `port`. Does it make sense?

Reply:
the ssh-copy-id command will default to 22 if you don't supply a port, so I don't think it is necessary?

Reply:
The command on the shell will default to 22 but not the ssCopyIdCommand() function. I used `None` as default port.
The question is who should "inject" the default port? Should BIT do it (if `proxy_host is None`) or should BIT not using any port if it is not set and let the shell command ssh-copy-id decide about default port?

Reply:
If I didn't set a port in BIT, then I would expect the default port to be decided by ssh-copy-id, not by BIT, as ssh-copy-id is likely the authority on what the defaults of its args should be.

Reply:
Shouldn't it then be the same for the regular `port=` argument?

Reply:
Damn... Now I brock BIT. :rofl:
I tried to leave the (proxy)port field empty to allow the SSH command itself to decide about the default port. But having an empty port value in the config break so many other parts of BIT. The code is not robust enough for this. I could fix it but it is to much and to risky. I will role back for this PR and explicit use "22" as default port specified by BIT. So BIT will always use a port value.

Reply:
ha!
Yes seems like it is probably better to be consistent with the BIT codebase than try to introduce a new way of handling it